:orphan:

GATK4: GetFilterMutectCalls
====================================================

*1 contributor Â· 3 versions*

Filter variants in a Mutect2 VCF callset.

FilterMutectCalls applies filters to the raw output of Mutect2. Parameters are contained in M2FiltersArgumentCollection and described in https://github.com/broadinstitute/gatk/tree/master/docs/mutect/mutect.pdf. To filter based on sequence context artifacts, specify the --orientation-bias-artifact-priors [artifact priors tar.gz file] argument one or more times. This input is generated by LearnReadOrientationModel.

If given a --contamination-table file, e.g. results from CalculateContamination, the tool will additionally filter on contamination fractions. This argument may be specified with a table for one or more tumor sample. Alternatively, provide a numerical fraction to filter with the --contamination argument. FilterMutectCalls can also be given one or more --tumor-segmentation files, which are also output by CalculateContamination.

Quickstart
-----------

    .. code-block:: python

       from janis_bioinformatics.tools.gatk4.filtermutectcalls.versions import Gatk4FilterMutectCalls_4_1_2

       wf = WorkflowBuilder("myworkflow")

       wf.step(
           "gatk4filtermutectcalls_step",
           Gatk4FilterMutectCalls(
               vcf=None,
               reference=None,
           )
       )
       wf.output("out", source=gatk4filtermutectcalls_step.out)
    

*OR*

1. `Install Janis </tutorials/tutorial0.html>`_

2. Ensure Janis is configured to work with Docker or Singularity.

3. Ensure all reference files are available:

.. note:: 

   More information about these inputs are available `below <#additional-configuration-inputs>`_.



4. Generate user input files for Gatk4FilterMutectCalls:

.. code-block:: bash

   # user inputs
   janis inputs Gatk4FilterMutectCalls > inputs.yaml



**inputs.yaml**

.. code-block:: yaml

       reference: reference.fasta
       vcf: vcf.vcf.gz




5. Run Gatk4FilterMutectCalls with:

.. code-block:: bash

   janis run [...run options] \
       --inputs inputs.yaml \
       Gatk4FilterMutectCalls





Information
------------


:ID: ``Gatk4FilterMutectCalls``
:URL: `https://software.broadinstitute.org/gatk/documentation/tooldocs/4.1.2.0/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php <https://software.broadinstitute.org/gatk/documentation/tooldocs/4.1.2.0/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php>`_
:Versions: 4.1.4.0, 4.1.3.0, 4.1.2.0
:Container: broadinstitute/gatk:4.1.2.0
:Authors: Hollizeck Sebastian
:Citations: TBD
:Created: 2019-09-09
:Updated: 2019-09-09



Outputs
-----------

======  ====================  =============================
name    type                  documentation
======  ====================  =============================
out     CompressedIndexedVCF  vcf containing filtered calls
======  ====================  =============================



Additional configuration (inputs)
---------------------------------

====================  ====================  ==================================  ==========  =============================================================================================================================================
name                  type                  prefix                                position  documentation
====================  ====================  ==================================  ==========  =============================================================================================================================================
vcf                   CompressedIndexedVCF  -V                                              vcf to be filtered
reference             FastaWithIndexes      -R                                              Reference sequence file
contaminationTable    Optional<File>        --contamination-table                           Tables containing contamination information.
segmentationFile      Optional<File>        --tumor-segmentation                            Tables containing tumor segments' minor allele fractions for germline hets emitted by CalculateContamination
statsFile             Optional<File>        --stats                                         The Mutect stats file output by Mutect2
readOrientationModel  Optional<File>        --orientation-bias-artifact-priors              One or more .tar.gz files containing tables of prior artifact probabilities for the read orientation filter model, one table per tumor sample
outputFilename        Optional<Filename>    -O                                           2
====================  ====================  ==================================  ==========  =============================================================================================================================================
